sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java

```diff
@@ -170,6 +171,51 @@ public String serializeList(List<?> list, CollectionFormat format) {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private static <T> T convertText(String value, Class<T> type) {
+        // Value is null or empty, return the default.
+        if (CoreUtils.isNullOrEmpty(value)) {
+            return null;
+        }
+
+        // Check the default value's type to determine how it needs to be converted.
+        Object convertedValue;
+        if (type.isAssignableFrom(byte.class)) {
+            convertedValue = Byte.parseByte(value);
+        } else if (type.isAssignableFrom(Byte.class)) {
+            convertedValue = Byte.valueOf(value);
+        } else if (type.isAssignableFrom(short.class)) {
+            convertedValue = Short.parseShort(value);
+        } else if (type.isAssignableFrom(Short.class)) {
+            convertedValue = Short.valueOf(value);
+        } else if (type.isAssignableFrom(int.class)) {
+            convertedValue = Integer.parseInt(value);
+        } else if (type.isAssignableFrom(Integer.class)) {
+            convertedValue = Integer.valueOf(value);
+        } else if (type.isAssignableFrom(long.class)) {
+            convertedValue = Long.parseLong(value);
+        } else if (type.isAssignableFrom(Long.class)) {
+            convertedValue = Long.valueOf(value);
+        } else if (type.isAssignableFrom(float.class)) {
+            convertedValue = Float.parseFloat(value);
+        } else if (type.isAssignableFrom(Float.class)) {
+            convertedValue = Float.valueOf(value);
+        } else if (type.isAssignableFrom(double.class)) {
+            convertedValue = Double.parseDouble(value);
+        } else if (type.isAssignableFrom(Double.class)) {
+            convertedValue = Double.valueOf(value);
+        } else if (type.isAssignableFrom(boolean.class)) {
+            convertedValue = Boolean.parseBoolean(value);
+        } else if (type.isAssignableFrom(Boolean.class)) {
+            convertedValue = Boolean.valueOf(value);
+        } else {
+            // Should this check if there are any String only constructors or static factories?
```

Comment:
like enums and expandable enums?

Reply:
Yeah that is one scenario

Reply:
I think supporting ExpandableStringEnum should be pretty straightforward. For others we may have to reflect on the class to get the constructor/ static method signatures, which I'm not sure if that's worth it.